pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -78,6 +78,16 @@
  * Main class for Pulsar broker service
  */
 public class PulsarService implements AutoCloseable {
+
+    private static final int EXECUTOR_THREAD_COUNT = Integer
+            .parseInt(System.getProperty("pulsar.executor-thread-count", "20"));
```

Comment:
I agree on that but I was having 2 concerns about it: 
 * We might end up many executor sizes in the `ServiceConfiguration` and it would be tricky to properly explain all of them
 * We typically provide a default value in the `conf/broker.conf`. If the default is dependent on the number of cores, that might be tricky.